src/transformers/tokenization_utils_base.py

```diff
@@ -1619,6 +1641,41 @@ def get_vocab(self) -> Dict[str, int]:
         """
         raise NotImplementedError()
 
+    def build_conversation_input_ids(
+        self,
+        conversation: Union[List[Dict[str, str]], ChatConversation],
+        prompt_config: Optional[PromptConfig] = None,
+        **kwargs,
+    ) -> List[int]:
+        if isinstance(conversation, ChatConversation):
+            conversation = conversation.messages
+
+        # priority: `prompt_config` argument > `model.prompt_config` (the default generation config)
+        if prompt_config is None:
+            prompt_config = self.prompt_config
+
+        # TODO How will this handle partial updates, such as when the kwargs only cover one role?
+        prompt_config.update(**kwargs)
+
+        if prompt_config.default_system_prompt is not None and (
+            len(conversation) == 0 or conversation.messages[0]["role"] != "system"
+        ):
+            conversation = [{"role": "system", "content": prompt_config.default_system_prompt}] + conversation.messages
+
+        dialog_tokens: List[int] = []
+
+        for message in conversation:
+            role = message["role"]
+            message_prefix = prompt_config.string_prefixes.get(role, "")
+            message_suffix = prompt_config.string_suffixes.get(role, "")
+            message_prefix_tokens = prompt_config.token_prefixes.get(role, [])
+            message_suffix_tokens = prompt_config.token_suffixes.get(role, [])
```

Comment:
If we only have 3 roles, they could be `@properties` in the `PromptConfig` wdyt?